ddtrace/tracer/tracer.go

```diff
@@ -45,6 +46,13 @@ type tracer struct {
 	// stopped is a channel that will be closed when the worker has exited.
 	stopped chan struct{}
 
+	// stopping synchronizes calls to Stop() to ensure the tracer is stopped
+	// exactly once.
+	stopping sync.Mutex
```

Comment:
Since you've now added the waitgroup (which is much nicer), we could reduce the ugliness here by using a construct that is more appropriate for exactly what we're doing here: https://golang.org/pkg/sync/#Once.

Basically I'm suggesting if it would work to replace this field with `stopOnce sync.Once` and remove both `stopping` and `stopped` and just do this:

```
func (t *tracer) Stop() {
    t.stopOnce.Do(func() {
        close(t.exitChan)
        t.wg.Wait()
    })
}
```
What do you think? Would it work? If yes, it seems much cleaner and nicer.

Reply:
Indeed. You’re right. I guess stopped could become a boolean, but let’s not drag this out anymore because it’s unrelated to this PR. Resolving.